core/trino-main/src/test/java/io/trino/server/security/oauth2/TestingHydraIdentityProvider.java

```diff
@@ -94,19 +102,34 @@ public void start()
         loginAndConsentServer.start();
         URI loginAndConsentBaseUrl = loginAndConsentServer.getBaseUrl();
 
+        certDir = new File(Resources.getResource("cert/localhost.pem").getPath()).getParentFile().toPath();
+        // generate a cert with Docker's host in SAN
+        String ipAddress = DockerClientFactory.instance().dockerHostIpAddress();
+        if (!ipAddress.equals("localhost")) {
+            certDir = Files.createTempDirectory("cert",
+                    PosixFilePermissions.asFileAttribute(EnumSet.allOf(PosixFilePermission.class)));
+            int exitCode = new ProcessBuilder()
+                    .command("./generate.sh", certDir.toString() + "/localhost.pem", "IP.2=" + ipAddress)
```

Comment:
Calling an external process should be a last resort. Use `io.airlift.security.cert.CertificateBuilder` instead. See example: https://github.com/airlift/airlift/blob/3d103a18ed91cdd80e0093041d246b9660c2e4ac/http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java#L452-L459

Reply:
Thanks! I was looking for something like that but my googling skills failed me and I didn't want to spend too much time figuring this out on my own.

Reply:
I'll wait before I make these changes until I get a confirmation whether we want to continue with this or just skip these tests on arm64.